modules/signals/entities/src/models.ts

```diff
@@ -0,0 +1,40 @@
+import { Signal } from '@angular/core';
+
+export type EntityId = string | number;
+
+export type EntityMap<Entity> = Record<EntityId, Entity>;
+
+export type EntityState<Entity> = {
+  entityMap: EntityMap<Entity>;
+  ids: EntityId[];
```

Comment:
Thanks for the review Tim! 🙏 

Yes, I was aware of this behavior. I was wondering if it's worth defining a different type for `EntityIds` as you mentioned because both `string` and `number` can be used to get the record property value by key.

For example, this code will compile:

```ts
const record: Record<number, number> = {
  '1': 1,
};
```

although we specified that record keys can be only numbers, because in JavaScript this works:

```ts
const record = { 1: 1 };

console.log(record[1]); // logs 1
console.log(record['1']); // logs 1
```

Let me know if you think this should be changed and I'll add the `EntityIds` type:

```ts
type EntityIds = string[] | number[];
```

cc @brandonroberts

Reply:
I'm fine with both options, I added this comment just to be sure that it was intended.

Reply:
I think allowing strings as record keys should be supported for use cases such as uuids.